account_invoice_pricelist/views/account_invoice_view.xml

```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+    <record id="view_account_move_customer_form" model="ir.ui.view">
+        <field name="model">account.move</field>
+        <field name="inherit_id" ref="account.view_move_form" />
+        <field name="arch" type="xml">
+            <button name="action_invoice_sent" position="before">
+                <button
+                    type="object"
+                    name="button_update_prices_from_pricelist"
+                    groups="product.group_product_pricelist"
+                    string="Update prices"
+                    attrs="{'invisible': ['|', '|', ('pricelist_id', '=', False), ('state', 'not in', ['draft']), ('move_type', 'not in', ['out_invoice', 'out_refund', 'in_invoice', 'in_refund'])]}"
+                    help="Update price in lines from the pricelist"
+                />
+            </button>
+            <field name="currency_id" position="attributes">
+                <attribute name="force_save">1</attribute>
+                <attribute
+                    name="attrs"
+                >{'readonly': [('pricelist_id', '!=', False)]}</attribute>
+            </field>
+            <xpath expr="//div[field[@name='journal_id']]" position="after">
+                <field
+                    name="pricelist_id"
+                    attrs="{'invisible': [('move_type', 'not in', ['out_invoice', 'out_refund', 'in_invoice', 'in_refund'])]}"
```

Comment:
@hildickethan-S73 you may face an error when adding **groups="product.group_product_pricelist"** to this field.
Fix it by adding one more field:
<field name="pricelist_id" groups="!product.group_product_pricelist" invisible="1"/>

I learn this from [sale_order_views.xml#L279](https://github.com/odoo/odoo/blob/16.0/addons/sale/views/sale_order_views.xml#L279)

Reply:
I skipped over this one by accident :sweat_smile: Yes it did error without the extra field because it is used in the domain above it. I added both so that the group covers all new fields now